xql/src/xql/apply.py

```diff
@@ -85,79 +84,31 @@ def parse(a: t.Union[xr.DataArray, str], b: t.Union[xr.DataArray, str]) -> t.Tup
     return a, b
 
 
-def evaluate(a: t.Union[xr.DataArray, str], b: t.Union[xr.DataArray, str], operator: str) -> xr.DataArray:
+def apply_orderby(e: exp.Order, df: pd.DataFrame) -> pd.DataFrame:
     """
-    Evaluate the expression 'a operator b' using NumPy arrays.
+    Apply ORDER BY clause to the DataFrame.
 
-    Parameters:
-    - a (Union[xr.DataArray, str]): The first input value.
-    - b (Union[xr.DataArray, str]): The second input value.
-    - operator (str): The operator to be applied.
-
-    Returns:
-    - xr.DataArray: The result of the evaluation.
-    """
-    a, b = parse(a, b)
-    return operate[operator](a, b)
-
-
-def inorder(expression: exp.Expression, ds: xr.Dataset) -> xr.DataArray:
-    """
-    Evaluate an expression using an xarray Dataset and return the result.
-
-    Parameters:
-    - expression (exp.Expression): The expression to be evaluated.
-    - ds (xr.Dataset): The xarray Dataset used for evaluation.
+    Args:
+    - e (exp.Order): Parsed ORDER BY expression.
+    - df (pd.DataFrame): DataFrame to be sorted.
 
     Returns:
-    - xr.DataArray: The result of evaluating the expression on the given dataset.
+    - pd.DataFrame: Sorted DataFrame based on the ORDER BY clause.
     """
+    orderby_columns = []
+    orderby_columns_order = []
 
-    if(expression.key == "identifier"):
-        return ds[expression.args['this']]
-
-    if(expression.key == "literal"):
-        return expression.args['this']
-
-    args = expression.args
+    # Extract columns and sorting orders from the parsed ORDER BY expression
+    for el in e.expressions:
+        orderby_column = el.find(exp.Column).find(exp.Identifier).this
+        order = not bool(el.args['desc'])  # Descending if desc=False
+        orderby_columns.append(orderby_column)
+        orderby_columns_order.append(order)
 
-    left = args['this']
-    right = None
-    if 'expression' in args:
-        right = args['expression']
+    # Sort the DataFrame based on the extracted columns and orders
+    df = df.sort_values(orderby_columns, ascending=orderby_columns_order)
```

Comment:
Also found out that sqlglot does not throw any error in case you provide extra columns in the order by clause (that are not even present in the table), it will add that column to the expression.

Example query 
```
query = "Select a, b from my_table order by a ASC, b, c";
```

You can try to print out the expression tree using this query. (Note column 'c' is not in our table)

I am not sure how pandas sort_values will handle that, but i case it doesn't it should be handled by us.

Reply:
For now let's proceed with the constraint that `orderby` can only be applied either on coordinates or selected variables.